src/components/hooks/useElementPosition.js

```diff
@@ -0,0 +1,37 @@
+import {useCallback, useEffect, useState} from 'react';
+
+/**
+ * Custom hook that returns an element reference and the position of the
+ * element it refers to.
+ *
+ * @param {object} elementRef  Reference to the element
+ *
+ * @returns {DOMRect} Element position
+ */
+export default function useElementPosition(elementRef) {
+  const [position, setPosition] = useState({});
+  const target = elementRef.current;
+
+  const handleSetSize = useCallback(() => {
+    if (target) {
+      setPosition(target.getBoundingClientRect());
+    }
+  }, [target]);
+
+  useEffect(() => {
+    handleSetSize();
```

Comment:
because this executes after each render, we need to prevent running setPosition if the position has not changed:
```js
const handleSetSize = useCallback(() => {
  if (!target) {
    setPosition(null);
  } else {
    const rect = target.getBounding();
    if (!position || position.top !== rect.top || position.left !== rect.left || ...) {
      setPosition(rect);
    }
  }
});